CONTRIBUTING.md

```diff
@@ -1,23 +1,48 @@
 # How To Contribute To Beaker
 
+Contributions are welcomed. Simple bug fixes and minor enhancements will usually be accepted. Larger features should be discussed with a team member before you invest in developing them with the expectation that they will be merged.
+
+## Recommended Tools
+
+It may be necessary to test Beaker against multiple versions of Ruby. The maintainers use [`rbenv`](https://github.com/rbenv/rbenv) to manage multiple Ruby versions; you can install it with Homebrew. You'll also want [`rbenv-bundler`](https://github.com/carsomyr/rbenv-bundler) to keep Gem dependencies from conflicting.
+
 ## Getting Started
 
+Beaker does not use GitHub Issues, but an internal ticketing system running Jira that interfaces with other services. To be accepted by the maintainers, changes must follow this workflow and tagging scheme. See [ticket process doc](docs/concepts/ticket_process.md) for a
+
 * Create a [Jira account](http://tickets.puppetlabs.com)
 * Make sure you have a [GitHub account](https://github.com/signup/free)
-* Submit a ticket for your issue, assuming one does not already exist.
+* Submit a ticket for your issue on Jira, assuming one does not already exist.
   * Clearly describe the issue including steps to reproduce when it is a bug.
-  * File a ticket in the [BKR project](https://tickets.puppetlabs.com/issues/?jql=project%20%3D%20BKR)
+  * File the ticket in the [BKR project](https://tickets.puppetlabs.com/issues/?jql=project%20%3D%20BKR)
 * Fork the [Beaker repository on GitHub](https://github.com/puppetlabs/beaker)
 
-## Making Changes
+## Installation/Setup for Development
 
-* Create a topic branch from your fork of [puppetlabs/beaker](https://github.com/puppetlabs/beaker). 
-  * Please title the branch after the beaker ticket you intend to address, ie `BKR-1234`.
-* Make commits of logical units.
-* Check for unnecessary whitespace with `git diff --check` before committing.
-* Make sure your commit messages are in the proper format.
+While most users will use Beaker as a Gem installed from some repository, you will need a live repo to work with. Here's how to configure Beaker and its dependencies so you can start contributing.
 
-````
+* Clone your fork of Beaker
+* Install Beaker's dependencies into `vendor/bundle`:
+  ```console
+    $ bundle install --path vendor/bundle
+  ```
+  * Installing the dependencies globally ~~may~~ *will probably* cause conflicts and is not recommended.
+  * Please use `vendor/bundle`, not `_vendor` or `.vendor`.
+* Ensure the spec tests pass:
+  ```console
+    $ rake test:spec # assuming you have rbenv-bundler
+    # or
+    $ bundle exec rake test:spec # if you're *sure* your dependencies are tidy
+  ```
+
+## Making Changes
+
+* Create a topic branch from your fork of [puppetlabs/beaker](https://github.com/puppetlabs/beaker).
+  * Please title the branch after the beaker ticket you intend to address, i.e. `BKR-1234`.
```

Comment:
never thought the branch mattered. Did you end up running that test to confirm it does?

Reply:
Branch naming has meaning, but the functionality is actually broken in the integration. Basically, the name of the branch gets picked up and auto-linked into the Jira ticket as a "Branch", but doesn't understand which repository that branch is connected to. See attached and note that the branch attempts to link to the main repository, where no such branch exists.
![image](https://user-images.githubusercontent.com/278782/41123850-e3bffcf2-6a54-11e8-8d67-072b0c26dafe.png)